iceoryx_hoofs/include/iceoryx_hoofs/internal/objectpool/objectpool.hpp

```diff
@@ -423,7 +423,8 @@ class ObjectPool
         return m_cellInfo[index].data;
     }
 
-  private:
+  protected:
+    // protected for unit tests
```

Comment:
This file was not yet re-worked according to our coding guidelines. Implementation and header file shall always be separate, also for templated classes. See an example [hpp](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp) and [inl](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl).

@r7vme If you're feeling adventurous today, I'd appreciate it if you do the split in this PR. One thing less to fix in future :)